### ISSUE TITLE
Visibility classes not working when inlining

### ISSUE THREAD

**Issue by StijnKeppens** [foundation/foundation-emails#857]
The visibility classes work fine when developing in "run start" but hide-for-large seems to fail when running "npm run build"

**How can we reproduce this bug?**
Write out the HTML (or Inky code) that causes the issue.

    
      <row class="hide-for-large">
        <columns  small ="12" large="3">
          <center>
            <img src="http://placehold.it/250/250" alt="">
          </center>
        </columns>
        <columns small ="12" large="9">
          <h6><strong>1. Bescherm je schoenen!</strong></h6>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </columns>
      </row>
      

    
    <row class="show-for-large">
      <columns large="7" small ="12">
        <h6><strong>2. Reinig je schoenen</strong></h6>
        <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
        </p>
      </columns>
      <columns large="5" small ="12">
        <center>
          <img src="http://placehold.it/250/250" alt="">
        </center>
      </columns>
    </row>
    

**What email clients does this happen in?**

Previewing the email in chrome, but the issue is there when sending the email to any of the browsers I've tested.

**Comment by erickreutz** [comment]
Also having this problem.

**Comment by erickreutz** [comment]
@StijnKeppens https://github.com/zurb/foundation-emails/commit/fdf6b4ee7e800c2cac4f3917003f748e59d85451#diff-4573bfa5a91c03813ad5aea9e1321cd1

**Comment by StijnKeppens** [comment]
Thx @erickreutz 
Did changing that one line in the _visibility.scss-file solve the issue for you? Because I did it but doesn't seem to have any effect...

**Comment by erickreutz** [comment]
Yes, try checking the README. It says to use a wrapper class around the content. https://github.com/zurb/foundation-emails/commit/fdf6b4ee7e800c2cac4f3917003f748e59d85451#diff-6b997f69dd51fc039b600a8baa51b8f5R19

**Comment by StijnKeppens** [comment]
Hmm, I’ve updated _visibility.scss (node_modules/foundation-emails/scss/components/_visibility.scss) but when running in build-mode, I can still see display: none!important; in the hide-for-large-class. Am I updating the wrong file?


Groet,
Stijn
0499 94 51 53

**Comment by stevesmename** [comment]
Does my comment in #694 solve your issue? I had put a -custom fix for this, and still works today. Yes a wrapper is needed for .hide-for-large to work. Note, I personally don't recommend messing with the source scss files as you basically just forked (however, Zurb F4E is pretty stale when it comes to stable releases).

**Comment by StijnKeppens** [comment]
Hi Steve,

Thanks. I've tried just that, changed node_modules/foundation-emails/scss/components/_visibility.scss but when I run pm run build, I can still find the display:none!importent; in the styling.
So either I’m modifying the wrong file, either there’s some step I’ve overlooked...

Regards,
Stijn

**Comment by stevesmename** [comment]
Hi @StijnKeppens - I do recall running into the same issue as you regarding the display:none!important; however this was avoided by using the custom scss code I include in #694 otherwise we would need to apply the patch every time we decided to download foundation for emails for various projects/clients. The recommended approach instead allows the ability to quickly apply a local foundation-fixes.scss file which fixes functionality with ease. I also know that you need to use a div wrapper when applying the .hide-for-large to work. 

So for your product 1, the following...

```
<!-- product 1 : display only on desktop-->
<div class="hide-for-large or hide-for-large-custom">
  <row>
    <columns  small ="12" large="3">
      <center>
        <img src="..." alt="">
      </center>
    </columns>
    <columns small ="12" large="9">
      <h6><strong>1. Bescherm je schoenen!</strong></h6>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </columns>
  </row>
</div>
  <!-- product 1 -->
```

**Comment by StijnKeppens** [comment]
Hi Steve,
Thanks for your help. I’ve tried this, but when inlining, I can still see the display:none!important; in my inspector window. I’had updated the_visibility.scss-file but maybe that’s not enough… Should I rebuild something or so to make sure these changes get applied?

Groet,
Stijn
0499 94 51 53

**Comment by joeworkman** [comment]
Foundation for Emails has been superseded by [Inky v2](https://github.com/foundation/inky), a complete rewrite in Rust with improved Outlook support, hybrid output mode, bulletproof buttons, validation, and more. This issue will not be addressed in the legacy codebase.

If this issue persists in v2, please open a new issue at https://github.com/foundation/inky/issues
